pkg/statemachine/epoch_target.go

```diff
@@ -121,7 +122,16 @@ func (et *epochTarget) step(source nodeID, msg *msgs.Msg) *ActionList {
 	return et.activeEpoch.step(source, msg)
 }
 
+// Constructs and returns a NewEpoch message.
+// This message consists of the configuration for the new epoch
+// and a set of digests of acked EpochChange messages.
+// Note specifically that this NewEpoch message does not contain the payload of the EpochChange messages.
+// The recipient receives those directly from other nodes.
+// TODO: Verify that the comment above makes sense.
 func (et *epochTarget) constructNewEpoch(newLeaders []uint64, nc *msgs.NetworkState_Config) *msgs.NewEpoch {
+
+	// TODO: Check whether this filtering is indeed meaningful.
+	//       Also remove the check for nil when calling, in case this function cannot return nil any more.
```

Comment:
This function can still return nil, at line 144 or line 149 (in fact these are where the expected nil is likely to come from) so the caller should certainly still check for 'nil'.